extension/data/background/handlers/modqueue.js

```diff
@@ -1,9 +1,40 @@
 'use strict';
 
 const MODQUEUE_CACHE_TTL = 30;
+const MODQUEUE_CACHE_NAME = 'tb-modqueue-cache';
 
-// Object containing the queue cached data per subreddit
-const queueCache = new Map();
+/**
+ * Sets the modqueue cache for a given subreddit.
+ * @param subreddit string subreddit
+ * @param cacheObject object containing the cache
+ */
+function setQueueCache (subreddit, cacheObject) {
+    return new Promise(resolve => {
+        browser.storage.local.get({[MODQUEUE_CACHE_NAME]: {}}).then(result => {
+            result[subreddit] = cacheObject;
+            browser.storage.local.set({
+                [`${MODQUEUE_CACHE_NAME}`]: result,
+            }).then(resolve);
+        });
+    });
+}
```

Comment:
Writing this as an async function rather than explicitly returning a `new Promise` makes it a bit simpler to understand at first glance.

```suggestion
async function setQueueCache (subreddit, cacheObject) {
    const result = await browser.storage.local.get({[MODQUEUE_CACHE_NAME]: {}});
    result[subreddit] = cacheObject;
    await browser.storage.local.set({
        [MODQUEUE_CACHE_NAME]: result,
    });
}
```